### PR TITLE
Changeset replace advopts

### PR DESCRIPTION
**Pull request by brianhatchl**

### CODE REVIEW

**Comment by brianhatchl** [comment]
This adds a list of advanced options for `hoot changeset-derive-replacement` to the grail dataset picker dialog.

image

**Comment by brianhatchl** [comment]
Also moves some methods for adv opts toggle to form factory to allow it to be reused.

**Comment by brianhatchl** [comment]
Additionally adds a `title` tooltip to advanced options fields for #1712 

Screenshot from 2020-01-24 14-35-53

Screenshot from 2020-01-24 14-36-38

Screenshot from 2020-01-24 14-35-27

File: modules/Hoot/ui/modals/grailDatasetPicker.js [hunk 1]
```diff
@@ -28,23 +29,44 @@ export default class GrailDatasetPicker {
...
     }
 
+    updateSubmitButton() {
+        this.submitButton.attr( 'disabled', function() {
+                var n = d3.select('#refDataset').property('value');
+                return (n && n.length) ? null : true;
+            });
+    }
 
     handleSubmit(d) {
```
Comment (bobi7) [inline modules/Hoot/ui/modals/grailDatasetPicker.js]: can remove variable `d` now

File: modules/Hoot/ui/modals/grailDatasetPicker.js [hunk 2]
```diff
@@ -28,23 +29,44 @@ export default class GrailDatasetPicker {
...
             };
 
+
+
         }
         let formId = 'grailDatasetForm';
-        this.form  = new FormFactory().generateForm( 'body', formId, metadata );
+        this.form  = this.formFactory.generateForm( 'body', formId, metadata );
 
+        // this.formFactory.createToggle(this.form);
```
Comment (bobi7) [inline modules/Hoot/ui/modals/grailDatasetPicker.js]: remove

**Pull request merged**